src/types/provider/response.ts

```diff
@@ -58,9 +64,45 @@ export type Block = {
   transactions: TXN_HASH[];
 };
 
-export type GetTransactionResponse = TransactionWithHash;
+export interface MessageToL1 {
+  to_address: string;
+  payload: Array<string>;
+}
+
+export type RevertedTransactionReceiptResponse = {
+  type?: TransactionType | any; // RPC only // any due to RPC Spec issue
+  execution_status: TransactionExecutionStatus.REVERTED | any; // any due to RPC Spec issue
+  finality_status: TransactionFinalityStatus | any;
+  status?: TransactionStatus; // SEQ only
+  actual_fee: string;
+  block_hash?: string; // ?~ optional due to RPC spec issue
+  block_number?: BlockNumber; // ?~ optional due to RCP spec issue
+  transaction_hash: string;
+  transaction_index?: number; // SEQ only
+  messages_sent: Array<MessageToL1>; // SEQ Casted l2_to_l1_messages
+  events: any[];
+  revert_reason?: string; // SEQ Casted revert_error // ?~ optional due to RCP spec issue
+};
 
-export type GetTransactionReceiptResponse = Receipt | PendingReceipt;
+export type RejectedTransactionReceiptResponse = {
+  status: `${TransactionStatus.REJECTED}`;
+  transaction_failure_reason: {
+    code: string;
+    error_message: string;
+  };
+};
+
+export type GetTransactionReceiptResponseWoHelper =
```

Comment:
I suupose "wo" means "without", but what do you think of a different abbreviation, like: `GetTxReceiptResponseWithoutHelper` ?

Reply:
Made